datafusion/datasource-parquet/src/file_format.rs

```diff
@@ -930,12 +959,14 @@ pub async fn fetch_parquet_metadata(
     store: &dyn ObjectStore,
     meta: &ObjectMeta,
     size_hint: Option<usize>,
+    decryption_properties: Option<&FileDecryptionProperties>,
```

Comment:
This looks like a breaking change (and same for `fetch_statistics`). Should we introduce a new method with this parameter added to avoid breaking the API? It is documented as "subject to change" though so maybe this is ok...

Reply:
I guess this probably needs to change anyway to handle having an `encryption` feature.